server/core/include/user_validation_utilities.hpp

```diff
@@ -0,0 +1,63 @@
+#ifndef IRODS_USER_VALIDATION_UTILITIES_HPP
+#define IRODS_USER_VALIDATION_UTILITIES_HPP
+
+#include <optional>
+#include <string>
+#include <string_view>
+#include <utility>
+
+struct RsComm;
+
+namespace irods::user
+{
+    /// \brief Look up the user type of the given user
+    ///
+    /// \param[in/out] _comm Server connection handle
+    /// \param[in] _user_name Name of the user whose user type is being fetched
+    ///
+    /// \return The user type of the given user as stored in the catalog
+    ///
+    /// \throws irods::exception If _user_name does not exist
+    ///
+    /// \since 4.2.11
+    auto get_type(RsComm* _comm, const std::string_view _user_name) -> std::string;
```

Comment:
if `_comm` is expected to be valid, we should make this a reference instead of a pointer.

Reply:
Yeps